src/Bicep.Core/Semantics/SemanticModel.cs

```diff
@@ -45,17 +46,17 @@ public class SemanticModel : ISemanticModel
         private readonly Lazy<ImmutableArray<ResourceMetadata>> allResourcesLazy;
         private readonly Lazy<ImmutableArray<DeclaredResourceMetadata>> declaredResourcesLazy;
         private readonly Lazy<ImmutableArray<IDiagnostic>> allDiagnostics;
+        private readonly ConcurrentDictionary<Uri, ResultWithDiagnostic<AuxiliaryFile>> fileCache = new();
```

Comment:
Do we expect non-auxiliary files in this cache? If yes, should we rename the variable to be more explicit?